tools/autograph-edge/main.go

```diff
@@ -0,0 +1,194 @@
+		flag.StringVar(&cfgFile, "c", "autograph-edge.yaml", "Path to configuration file")
+		flag.StringVar(&token, "a", "dd095f88adbf7bdfa18b06e23e83896107d7e0f969f7415830028fa2c1ccf9fd", "auth token")
+		flag.StringVar(&filePath, "f", "/path/to/file", "path to file to sign")
+		flag.Parse()
+
+		body, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = conf.loadFromFile(cfgFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		headers := make(map[string]string)
+		headers["Authorization"] = token
+		resp, err := Handler(events.APIGatewayProxyRequest{
+			HTTPMethod: http.MethodPost,
+			Body:       base64.StdEncoding.EncodeToString(body),
+			Headers:    headers,
+		})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error: %v\n", err)
+			os.Exit(1)
+		}
+		if resp.StatusCode != http.StatusOK {
+			log.Fatal("signing failed")
+		}
+		signedBody, err := base64.StdEncoding.DecodeString(resp.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+		ioutil.WriteFile(filePath, signedBody, 06400)
+		log.Println("signed file written to", filePath)
+	}
+}
+
+// Handler receives requests from AWS API Gateway and processes them. The input body must
+// contain a base64 encoded file to sign, and the response body contains a base64 encoded
+// signed file. The Authorization header of the http request must contain a valid token.
+func Handler(r events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+	log.WithFields(log.Fields{
+		"remoteAddressChain": "[" + r.Headers["X-Forwarded-For"] + "]",
+		"method":             r.HTTPMethod,
+		"url":                r.Path,
+		"content-type":       r.Headers["content-type"],
+		"rid":                r.RequestContext.RequestID,
+	}).Info("request")
+
+	// some sanity checking on the request
+	if r.HTTPMethod != http.MethodPost {
+		log.WithFields(log.Fields{"rid": r.RequestContext.RequestID}).Error("invalid method")
+		return events.APIGatewayProxyResponse{StatusCode: http.StatusMethodNotAllowed, Body: errInvalidMethod.Error()}, errInvalidMethod
+	}
+	if len(r.Headers["Authorization"]) < 60 {
+		log.WithFields(log.Fields{"rid": r.RequestContext.RequestID}).Error("missing authorization header")
+		return events.APIGatewayProxyResponse{StatusCode: http.StatusUnauthorized, Body: errInvalidToken.Error()}, errInvalidToken
+	}
+	if len(r.Body) < 1 {
+		log.WithFields(log.Fields{"rid": r.RequestContext.RequestID}).Error("missing request body")
+		return events.APIGatewayProxyResponse{StatusCode: http.StatusBadRequest, Body: errMissingBody.Error()}, errMissingBody
+	}
+
+	decodedBody, err := base64.StdEncoding.DecodeString(r.Body)
+	if err != nil {
+		log.WithFields(log.Fields{"rid": r.RequestContext.RequestID}).Error(err)
+		return events.APIGatewayProxyResponse{StatusCode: http.StatusBadRequest, Body: err.Error()}, err
+	}
+
+	log.WithFields(log.Fields{"rid": r.RequestContext.RequestID}).Info("checking authorization token")
+	auth, err := authorize(r.Headers["Authorization"])
```

Comment:
Should we verify the authorization before handing the request body off to the decoder vs. after?

Reply:
good catch! fixed in https://github.com/mozilla-services/autograph/pull/72/commits/4fa775a1b769db1ba4bf3936b690f8ed82cf5f42